src/Raven.Server/Documents/ETL/Providers/S3/RowGroups.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Parquet.Data;
+using Raven.Server.Documents.ETL.Providers.SQL;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.ETL.Providers.S3
+{
+    public class RowGroups
+    {
+        public RowGroups(string name, string key)
+        {
+            TableName = name;
+            PartitionKey = key;
+            Groups = new List<RowGroup>();
+        }
+
+        public string TableName { get; set; }
+
+        public string PartitionKey { get; set; }
+
+        public Field[] Fields { get; private set; }
+
+        public List<RowGroup> Groups { get; }
+
+        private const int MaxItemsPerGroup = 50_000;
+
+        public void Add(ToS3Item item)
+        {
+            var group = GetCurrentGroup(item);
+            var propCount = item.Properties.Count;
```

Comment:
What happens when different items have different number of fields?
What happens if the order of the field isn't fixed?